### DISCUSSION TITLE
Why am I not able to run minitest.py on my machine locally ?

### DISCUSSION THREAD

**Discussion by gobbledy-gook** [open-quantum-safe/oqs-provider#512] [Q&A] [answered]
I want to use the `oqs` module in my python code. Now to do that I have seen the liboqs-python setup, tried setting it up and ran the `minitest.py` but I couldn't get successful connection on ports other than 6000 and 6054. I know that this is not the outcome I need. As discussed in #507 running `minitest.py` should give success on all the ports from 6000 to 6108.

To setup the liboqs-python I followed the steps as in DockerFile but locally on my machine. Following are the terminal commands I ran:

```
cd /opt
sudo git clone --depth 1 --branch main https://github.com/open-quantum-safe/liboqs && sudo git clone --depth 1 --branch main https://github.com/open-quantum-safe/liboqs-python.git 

cd /opt/liboqs
sudo mkdir build && cd build && sudo cmake -GNinja .. -DOQS_DIST_BUILD=ON -DBUILD_SHARED_LIBS=ON -DOQS_USE_OPENSSL=OFF && sudo ninja install

 cd /opt
sudo git clone --depth 1 --branch OQS-OpenSSL_1_1_1-stable https://github.com/open-quantum-safe/openssl.git && cd liboqs && sudo mkdir build-openssl && cd build-openssl && sudo cmake -G"Ninja" .. -DOQS_DIST_BUILD=ON -DBUILD_SHARED_LIBS=ON -DOQS_USE_OPENSSL=OFF -DCMAKE_INSTALL_PREFIX=/opt/openssl/oqs && sudo ninja install

brew install automake autoconf && cd /opt/openssl && LDFLAGS="-Wl,-rpath,/usr/local/lib" ./Configure shared darwin64-x86_64-cc -lm && sudo make ${MAKE_DEFINES} && sudo make install_sw

wget https://letsencrypt.org/certs/isrgrootx1.pem

```

then setting up the environment variable `PYTHONPATH=/opt/liboqs-python` 

```
cd /opt/liboqs-python && python3 setup.py install
```

then I tried to run `minitest.py`

**So what went wrong here?** My prediction is that `OQS-OpenSSL_1_1_1-stable https://github.com/open-quantum-safe/openssl.git` could cause an issue because of the unsupported version of openssl-1.1.1 but then when I run using the docker container of liboqs-python it runs perfectly fine.

**My goal is to be able to run `minitest.py` locally from my machine with the same results as I get when I run it from the docker container.**

Any help or guidance will be highly appreciated!!
Thanks!

**Comment by baentsch** [comment]
> My goal is to be able to run minitest.py locally from my machine with the same results as I get when I run it from the docker container.

Can you please be specific as to which docker container you mean?

**Reply by gobbledy-gook** [reply]
Apologies for the non-specifity. I intended to mean docker container of `openquantumsafe/python`

**Comment by baentsch** [comment, marked as answer]
> My prediction is that OQS-OpenSSL_1_1_1-stable https://github.com/open-quantum-safe/openssl.git could cause an issue because of the unsupported version of openssl-1.1.1 but then when I run using the docker container of liboqs-python it runs perfectly fine

Well, it may well do this -- but if you mean [this code](https://github.com/open-quantum-safe/liboqs-python/blob/main/docker/Dockerfile) (?) it indeed has not been updated in 2 years and is definitely outdated. That Docker image is in dire need of upgrading to OpenSSL3/oqsprovider along the lines of https://github.com/open-quantum-safe/oqs-demos/issues/182.

**Reply by gobbledy-gook** [reply]
XD No wonder why it wasn't working! Thanks for the support!

Alright, then it would be great to work on the integration of liboqs-python with the oqs-provider... Could you suggest a good starting point, @baentsch?

**Reply by baentsch** [reply]
> it would be great to work on the integration of liboqs-python with the oqs-provider... Could you suggest a good starting point

It'd be great if you'd be willing to take this on @gobbledy-gook ! The full-build sample of curl using oqs-provider may be a useful starting point: https://github.com/open-quantum-safe/oqs-demos/blob/main/curl/Dockerfile
